core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -2163,4 +2167,26 @@ public String toString() {
                     '}';
         }
     }
+
+    static class EnrichedRecordBatch {
+        private final RecordBatch batch;
+        private final int skippedBytes;
+
+        public EnrichedRecordBatch() {
+            this(null, 0);
+        }
+
+        public EnrichedRecordBatch(RecordBatch batch, int skippedBytes) {
+            this.batch = batch;
+            this.skippedBytes = skippedBytes;
+        }
+
+        public RecordBatch getBatch() {
```

Comment:
We don't use getters. So `getBatch` => `batch` and `getSkippedBytes` => `skippedBytes`.